### ISSUE TITLE
fix: Location.getLocation() not resolving when app location accouracy is not set to Precise

### ISSUE THREAD

**Issue by TouseefAQ** [Lyokone/flutterlocation#748]
When i turn off location accuracy from app setting in ios , getLocation doesn't resolve.Althought i have set location accuracy to LocationAccuracy.low from changeSetting methode.
Similary on Android 7 , user device location is turned off but IslocationServiceEnabled returns true every time.

It should ateast get approximate location.

**Steps To Reproduce**

Go to ios App setting
Turn off presise
And Try to call GetLocation()

**Tested on:**

- Android, 7.1.2 &  12
- iOS, Iphone 7 , os 15.5 

Flutter Doctor v

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.5, on macOS 11.3.1 20E241 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.67.1)
[✓] VS Code (version 1.66.2)
[✓] Connected device (2 available)
    ! Error: Anas iphone12 is not connected. Xcode will continue when Anas iphone12 is connected. (code -13)
    ! Error: iPhone is busy: Fetching debug symbols for iPhone. Xcode will continue when iPhone is finished. (code -10)
[✓] HTTP Host Availability

**Comment by troy379** [comment]
Same for me. Any thoughts on how to resolve this?

**Comment by TouseefAQ** [comment]
> Tried adding `_location.changeSettings(accuracy: LocationAccuracy.low);` as well ass `_location.changeSettings(accuracy: LocationAccuracy.reduced);`, but it didn't help :(

Exactly

**Comment by troy379** [comment]
Since I needed this functionality ASAP, and there's no feedback so far, I had to switch to the [geolocator library](https://pub.dev/packages/geolocator). Maybe this will help you 🙂

**Comment by TouseefAQ** [comment]
> Since I needed this functionality ASAP, and there's no feedback so far, I had to switch to the [geolocator library](https://pub.dev/packages/geolocator). Maybe this will help you 🙂

Yeah but geolocator responds late on iOS, especially on 15.5. What I have done for now is to set **location accuracy to accuracy.reduced** and always check for permission granted. If permission is not fully granted don't let the user enter the app until or unless permission is fully granted.

**Comment by TouseefAQ** [comment]
Still no update ?

**Comment by RottenSchnitzel** [comment]
I am also encountering this issue and it is a dealbreaker for me, it's concerning that there is not even an update on this yet. There are two issues open regarding the problem (this and #764) and I think they originate in the same bug.

I hope this gets fixed soon as I prefer this package to geolocator. Are there still no updates or temporary fixes?

**Comment by Lyokone** [comment]
Fixed in 9.0.0. On iOS, with app location accuracy turned off, the single reduced accuracy update was being dropped, so getLocation never resolved. That path was reworked in the Swift rewrite.

https://pub.dev/packages/location/versions/9.0.0

Closing. Reopen if you still see it on 9.0.0.
